src/OpenCSGRenderer.cc

```diff
@@ -123,10 +123,22 @@ void OpenCSGRenderer::renderCSGProducts(const CSGProducts &products, GLint *shad
 				colormode = ColorMode::MATERIAL;
 			}
 			
-			setColor(colormode, c.data(), shaderinfo);
 			glPushMatrix();
 			glMultMatrixd(csgobj.leaf->matrix.data());
-			render_surface(csgobj.leaf->geom, csgmode, csgobj.leaf->matrix, shaderinfo);
+			
+			if (setColor(colormode, c.data(), shaderinfo)) {
```

Comment:
I think it would be cleaner to move the alpha check out of `setColor()` and do it manually here.

Reply:
The problem is that setColor doesn't unconditionally set the color that you give it from this scope. It depends if the object is highlighted, etc.  Putting the check here would mean basically duplicating all the code of setColor.   This function previously had void return type, I figured a boolean return was the simplest way to get that info out of the function.  Would it be preferrable to return the full Color4f basecol instead?  Then check the alpha of that here?

Reply:
Yes, that's a cleaner solution.
Better would be to e.g. split out those calculations from `setColor()`, but that's probably out of scope for this PR.